### ISSUE TITLE
Correspondation between actual x-y coordinates from Vertex Editor and the coordinates in Attribute Table

### ISSUE THREAD

**Issue by DavinavanMeer** [qgis/QGIS#57122]
### Feature description

After a lot of back and forth between the people who gave us the coordinates and what it showed in Qgis (and a lot of googling later) I found out that there are two places you have to change the coordinates, before the actual location of the point is changed. In the attribute table there's place for X and Y, which is what you assign as X and Y when you import the CSV-file, but when you change the coordinates in that spot later, the actual coordinates don't change with it. You have to manually change the X and Y in the Vertex Editor as well, before the location changes. 
The issue with changing it with the Vertex Editor is that the value doesn't change in the attribute table either. So you always have to change it in both places manually (or with an extra plugin), which is kind of annoying and very error sensitive. 

What I'd love to see is a continues connection between the two values, so that you don't have to change it in both places yourself. It would be way less error sensitive and time consuming (especially if you have a lot of points). 

Right now working in Qgis version 3.30.2. Added a screenshot to show what I mean with both places.

![Screenshot vertex editor and attribute table](https://github.com/qgis/QGIS/assets/166698113/fc330152-4180-45d6-8629-7e5a12d442bf)

**Comment by antoniolocandro** [comment]
CSV is a ready only file, so I suspect you read it and created a point layer. If you want to alter the CSV you can change the coordinates in Notepad++ for example and would reflect in map canvas. Now if you imported the CSV converted to point and then move the point graphically then you will need to update the X and Y fields of course. For that to be automatic you can use something like detailed on this post https://www.reddit.com/r/QGIS/comments/mwzcoc/is_it_possible_to_automatically_update_an/#:~:text=Yes.,QGIS%20will%20recalculate%20the%20area.

Basically on feature update to automatically change x and y coordinates, you will need to use $x and $y.
